hummingbird/ml/_container.py

```diff
@@ -52,24 +52,57 @@ def __init__(self, input_names, output_names, operator_map, operators, extra_con
             extra_config: Some additional custom configuration parameter
         """
         super(PyTorchBackendModel, self).__init__()
-        self._input_names = input_names
-        self._output_names = output_names
+
+        # Define input \ output names.
+        # This is required because the internal variable names may differ from the original (raw) one.
+        # This may happen, for instance, because we force our internal naming to be unique.
+        self._input_names = []
+        self._output_names = []
+        map = {}
+        for op in operators:
+            for op_input in op.inputs:
+                for i_name in input_names:
+                    if op_input.raw_name == i_name and i_name not in map:
+                        map[op_input.raw_name] = op_input.full_name
+            if len(input_names) == 0:
+                break
+        for i_name in input_names:
+            self._input_names.append(map[i_name])
+
+        map = {}
```

Comment:
lines 61-70 are almost identical to 72-81. is there any way to combine?

Reply:
Yep, fixed.